spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RuntimeAppsController.java

```diff
@@ -143,10 +146,39 @@ public RuntimeAppsController(StreamDefinitionRepository streamDefinitionReposito
 						.collect(Collectors.toList())
 		).get();
 
+		enrichWithMetrics(statuses);
+
 		// finally, pass in pageable and tell how many items we have in all pages
 		return assembler.toResource(new PageImpl<>(statuses, pageable, deploymentIds.size()), statusAssembler);
 	}
 
+	private void enrichWithMetrics(List<AppStatus> statuses) {
+		List<ApplicationsMetrics> metricsIn = metricStore.getMetrics();
+		Map<String, ApplicationsMetrics.Instance> metricsInstanceMap = new HashMap<>();
+		for (ApplicationsMetrics am : metricsIn) {
+			for (ApplicationsMetrics.Application a : am.getApplications()) {
+				for (ApplicationsMetrics.Instance i : a.getInstances()) {
+					metricsInstanceMap.put(i.getGuid(), i);
+				}
+			}
+		}
+
+		for (AppStatus appStatus : statuses) {
+			Map<String, AppInstanceStatus> appInstanceStatusMap = appStatus.getInstances();
+			appInstanceStatusMap.forEach((k,appInstanceStatus) -> {
+				String trackingKey = appInstanceStatus.getAttributes().get("guid");
+				if (metricsInstanceMap.containsKey(trackingKey)) {
+					ApplicationsMetrics.Instance metricsAppInstance = metricsInstanceMap.get(trackingKey);
+					appInstanceStatus.getAttributes().put("metrics.integration.channel.input.receiveRate",
+							String.format("%.2f", metricsAppInstance.getIncomingRate()));
+					appInstanceStatus.getAttributes().put("metrics.integration.channel.output.sendRate",
+							String.format("%.2f", metricsAppInstance.getOutgoingRate()));
+				}
+			});
+		}
+
+	}
+
 	@RequestMapping("/{id}")
 	public AppStatusResource display(@PathVariable String id) {
```

Comment:
do you think it is worth adding a REST endpoint in `spring-cloud-dataflow-metrics-collector` that fetches the metrics for a given Stream and incorporate the status with the metrics for that stream here?